.github/workflows/comment-on-release.yml

```diff
@@ -0,0 +1,126 @@
+name: Comment on PRs in Release
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  pull-requests: write
+  contents: read
+
+jobs:
+  comment-on-prs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get previous release tag
+        id: previous_tag
+        run: |
+          # Get all tags sorted by version
+          CURRENT_TAG="${{ github.event.release.tag_name }}"
+          PREVIOUS_TAG=$(git tag --sort=-version:refname | grep -A 1 "^${CURRENT_TAG}$" | tail -n 1)
+
+          # If there's no previous tag, use the first commit
+          if [ -z "$PREVIOUS_TAG" ] || [ "$PREVIOUS_TAG" = "$CURRENT_TAG" ]; then
+            PREVIOUS_TAG=$(git rev-list --max-parents=0 HEAD)
+          fi
+
+          echo "previous_tag=${PREVIOUS_TAG}" >> $GITHUB_OUTPUT
+          echo "Found previous tag/commit: ${PREVIOUS_TAG}"
+
+      - name: Get merged PRs between releases
+        id: get_prs
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const currentTag = '${{ github.event.release.tag_name }}';
+            const previousTag = '${{ steps.previous_tag.outputs.previous_tag }}';
+
+            console.log(`Finding PRs between ${previousTag} and ${currentTag}`);
+
+            // Get commits between previous and current release
+            const comparison = await github.rest.repos.compareCommits({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              base: previousTag,
+              head: currentTag
+            });
+
+            const commits = comparison.data.commits;
+            console.log(`Found ${commits.length} commits`);
+
+            // Extract PR numbers from commit messages
+            const prNumbers = new Set();
+            const prPattern = /#(\d+)/g;
+
+            for (const commit of commits) {
+              const message = commit.commit.message;
+              const matches = message.matchAll(prPattern);
```

Comment:
Instead of this Regex which seems like it could find a bunch of false positives (e.g. if it says "Fixes #1234" it would match this), can we use the `listPullRequestsAssociatedWithCommit` API GH exposes?

https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-pull-requests-associated-with-a-commit